### PR TITLE
AddCampaignDraft example.

### PR DESCRIPTION
**Pull request by nwbirnie**
Ported from https://github.com/googleads/google-ads-python/pull/251

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    try (CampaignDraftServiceClient campaignDraftServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignDraftServiceClient()) {
+      // Issues the mutate request.
+      MutateCampaignDraftsResponse response = campaignDraftServiceClient
+          .mutateCampaignDrafts(
+              String.valueOf(customerId), ImmutableList.of(opBuilder.build()));
+
+      // Displays the result.
+      for (MutateCampaignDraftResult result : response.getResultsList()) {
+        System.out.printf("Created campaign draft: %n", result.getResourceName());
```
[the remark this answers is not in the document]
Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java]: We don't really have a consistent way of handling this:
- [Quoted](https://github.com/googleads/google-ads-java/blob/6a9a9bc428647683994e9d0b0b818c1597695671/google-ads-examples/src/main/java/com/google/ads/googleads/examples/basicoperations/UpdateKeyword.java#L129-L131)
- [Newline](https://github.com/googleads/google-ads-java/blob/6a9a9bc428647683994e9d0b0b818c1597695671/google-ads-examples/src/main/java/com/google/ads/googleads/examples/basicoperations/AddCampaigns.java#L178)

I've updated to quote them here.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import com.google.ads.googleads.v3.services.MutateCampaignDraftsResponse;
+import com.google.ads.googleads.v3.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Adds a campaign draft for a campaign. Make sure you specify a campaign that has a non-shared
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:35]: Should the comment about the campaign with a non-shared budget be on the campaign param?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:88]: Add params to javadoc

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId) {
+    // Creates an operation to create a new campaign draft.
+    CampaignDraftOperation.Builder opBuilder = CampaignDraftOperation.newBuilder();
+
+    // Sets the newly created campaign draft to the provided base campaign ID.
+    opBuilder
+        .getCreateBuilder()
+        .setBaseCampaign(StringValue.of(ResourceNames.campaign(customerId, baseCampaignId)))
+        .setName(StringValue.of("Campaign Draft #" + UUID.randomUUID()));
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:97]: In other examples we use System.currentTimeMillis()

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId) {
+    // Creates an operation to create a new campaign draft.
+    CampaignDraftOperation.Builder opBuilder = CampaignDraftOperation.newBuilder();
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:91]: Why not just create the entire CampaignDraftOperation? Do we need a builder here?

Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:91]: Builder is required since we call `getCreateBuilder()` so the result of chaining `build()` to the end would be `CampaignDraft` not `CampaignDraftOperation`.

That said, since this is hard to follow I've swapped it out.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java (hunk 2 above)
Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:35]: This matches the golden

https://github.com/googleads/google-ads-python/pull/251/files

I think it's a reasonable caveat that's worth including here. I've included it below in the javadoc as well.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId) {
+    // Creates a CampaignDraft object with the provided base campaign ID.
+    CampaignDraft draft = CampaignDraft.newBuilder()
+        .setBaseCampaign(StringValue.of(ResourceNames.campaign(customerId, baseCampaignId)))
+        .setName(StringValue.of("Campaign Draft #" + System.currentTimeMillis()))
+        .build();
+
+    // Creates an operation to create the draft campaign.
+    CampaignDraftOperation opBuilder = CampaignDraftOperation.newBuilder().setCreate(draft).build();
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:104]: Change var name to 'operation'

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+    // Creates a service client to connect to the API.
+    try (CampaignDraftServiceClient campaignDraftServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignDraftServiceClient()) {
+      // Issues the mutate request.
+      MutateCampaignDraftsResponse response =
+          campaignDraftServiceClient.mutateCampaignDrafts(
+              String.valueOf(customerId), ImmutableList.of(opBuilder));
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/AddCampaignDraft.java:112]: Change var name to 'operation'
